packages/venia-ui/lib/components/RichContent/PageBuilder/factory.js

```diff
@@ -0,0 +1,50 @@
+import React, { Suspense } from 'react';
+import Missing from './missing';
+import { contentTypesConfig, Lazy } from './config';
+
+/**
+ * Render a content type
+ *
+ * @param Component
+ * @param data
+ * @returns {*}
+ */
+const renderContentType = (Component, data) => {
+    return (
+        <Component {...data}>
+            {data.children.map((childTreeItem, i) => (
+                <ContentTypeFactory key={i} data={childTreeItem} />
+            ))}
+        </Component>
+    );
+};
+
+/**
+ * Create an instance of a content type component based on configuration
+ *
+ * @param data
+ * @returns {*}
+ * @constructor
+ */
+const ContentTypeFactory = ({ data }) => {
+    const contentTypeConfig = contentTypesConfig[data.contentType];
+
+    if (contentTypeConfig) {
+        const PageBuilderComponent = contentTypeConfig.component;
+
+        // If we're lazy loading add some suspense
+        if (contentTypeConfig.load === Lazy) {
```

Comment:
Cool. 👍

Reply:
I initially tried to do a type assertion of `component` to determine if it wanted to be lazy or not. I couldn't get this working, but if you have any ideas so we don't have to have a somewhat pointless `load` argument that'd be great.

Reply:
Can you do it with `static` properties on the component class itself?